### DISCUSSION TITLE
Don't run on tag creation?

### DISCUSSION THREAD

**Discussion by SoftCreatR** [community/community#25615] [Actions]
<p>It seems, that <code>on: [ push ]</code> get’s triggered, whenever you make changes and also, whenever a tag/release is being created. Is there a way to prevent this? I don’t see a need to process a commit, that has already been processed.</p>

**Comment by kingthorin** [comment, marked as answer]
<p>If you want to run on pushes but ignore all tags you’d need something like:</p>
<pre><code class="lang-auto">on:
  push:
    tags-ignore:
      - '**'
</code></pre>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" target="_blank" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<blockquote>
<p>The <code>branches</code> , <code>branches-ignore</code> , <code>tags</code> , and <code>tags-ignore</code> keywords accept glob patterns that use the <code>*</code> and <code>**</code> wildcard characters to match more than one branch or tag name. For more information, see the “<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet" rel="noopener nofollow ugc">Filter pattern cheat sheet</a>.”</p>
</blockquote>

**Comment by SoftCreatR** [comment]
<p>Ty, I’ll try that. Seems to do, what I’m looking for.</p>
<p>EDIT:</p>
<p>Works well.</p>

**Reply by Zordid** [reply]
The post below yours states it does not work as the trigger would never fire... who's right?

**Comment by BrightRan** [comment]
<p><a class="mention" href="/softcreatr">@SoftCreatR</a>, <a class="mention" href="/kingthorin">@kingthorin</a>,</p>
<p>According to the introduction in the docs:</p>
<blockquote>
<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="noopener nofollow ugc">If you define only <strong>tags</strong> or only <strong>branches</strong>, the workflow won’t run for events affecting the undefined Git ref.</a></p>
</blockquote>
<p>If you do not want the push event is triggered when pushing/creating tags, you can only set branch filters (<strong>branches</strong> or <strong>branches-ignore</strong>) for the push event in your workflow.</p>
<pre><code class="lang-yaml">on:
  push:
    branches:
      - 'some branch-filters-patterns'
</code></pre>
<p>OR</p>
<pre><code class="lang-yaml">on:
  push:
    branches-ignore:
      - 'some branch-filters-patterns'
</code></pre>
<br>
<blockquote>
<p><a class="mention" href="/kingthorin">@kingthorin</a>,<br>
If you want to run on pushes but ignore all tags you’d need something like:</p>
<pre><code class="lang-auto">on:
 push:
   tags-ignore:
     - '**'
</code></pre>
</blockquote>
<p>This configuration will cause the workflow to never be triggered. This is equivalent to disabling the workflow, and this workflow will become meaningless. If so, why don’t you directly delete the workflow file?<br>
The following configurations will cause the same result too.</p>
<ul>
<li>Only set tag filters (<strong>tags</strong>) with the negative pattern (’<strong>!**</strong>’).<pre><code class="lang-yaml">on:
  push:
    tags:
      - '!**'
</code></pre>
</li>
<li>Only set branch filters (<strong>branches</strong>) with the negative pattern (’<strong>!**</strong>’).<pre><code class="lang-yaml">on:
  push:
    branches:
      - '!**'
</code></pre>
</li>
<li>Only set branch filters ( <strong>branches-ignore</strong>) with the positive pattern (’<strong>**</strong>’).<pre><code class="lang-yaml">on:
  push:
    branches-ignore:
      - '**'
</code></pre>
</li>
</ul>

**Comment by SoftCreatR** [comment]
<p>Good to know. I’ll update the Workflow accordingly.</p>

**Comment by kingthorin** [comment]
<p>Fair enough: Any branch, ignore all tags.</p>
<pre><code class="lang-auto">on:
  push:
    branches:
    - '**'
    tags-ignore:
      - '**'
</code></pre>
<p>Although confusingly and contrary to the bit <a class="mention" href="/brightran">@brightran</a> quoted, <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet</a> states:</p>
<blockquote>
<p><code>'**'</code> `` Matches all branch and tag names. This is the default behavior when you don’t use a <code>branches</code> or <code>tags</code> filter</p>
</blockquote>
<p>Which to me implies that if you don’t specify branches then all branches are included… Right? If you don’t include a branch filter then you’re only specifying a tag filter or not specifying a filter at all.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/kingthorin">@kingthorin</a>, <a class="mention" href="/softcreatr">@SoftCreatR</a>,</p>
<blockquote>
<p><strong>’**’</strong>:  Matches all branch and tag names. This is the default behavior when you don’t use a <strong>branches</strong> or <strong>tags</strong> filter.</p>
</blockquote>
<p>This description includes 3 main situations below:<br>
(Due to the reason I mentioned above, not consider using this pattern for <strong>tags-ignore</strong> and <strong>branches-ignore</strong>.)</p>
<ol>
<li>
<p>Only set tag filters ( <strong>tags</strong> ) with the negative pattern (<strong>’**’</strong>). The workflow is only triggered when you push tags. Pushing commits to branches can’t trigger the workflow.</p>
<pre><code class="lang-yaml">on:
  push:
    tags:
      - '**'
</code></pre>
</li>
<li>
<p>Only set branch filters ( <strong>branches</strong> ) with the negative pattern (<strong>’**’</strong>). The workflow is only triggered when you push commits to branches. Pushing tags can’t trigger the workflow.</p>
<pre><code class="lang-yaml">on:
  push:
    branches:
      - '**'
</code></pre>
</li>
<li>
<p>Not set any tag filters and branch filters. The workflow will be triggered by both pushing commits to branches and pushing tags.</p>
<pre><code class="lang-yaml">on:
  push:
</code></pre>
<p>This configuration is equivalent to:</p>
<pre><code class="lang-yaml">on:
  push:
    tags:
      - '**'
    branches:
      - '**'
</code></pre>
</li>
</ol>
<p>Situation 1 and 2 correspond to the description “<strong>Matches all branch and tag names</strong>”.<br>
Situation 3 corresponds to the description “<strong>This is the default behavior when you don’t use a branches or tags filter</strong>”.</p>

**Reply by johan-boule** [reply]
negative pattern (’**’) ?

**Comment by kingthorin** [comment]
<p>Okay, becoming clearer. Should the pattern in 1 and 2 be ‘!**’ (bang double asterisk)? Or, is the word ‘negative’ misplaced? (Or is there something else subtle I’m missing?)</p>

**Comment by Simran-B** [comment]
<p>Regarding situation 3, should the wording be slightly changed to make it clearer?</p>
<blockquote>
<p>“This is the default behavior when you don’t use a branches <strong>nor</strong> tags filter”</p>
</blockquote>

**Comment by opyate** [comment]
<p>Hey all, I looked through the replies, but not sure if the OP’s original query has been answered: preventing <code>on:create</code> on tags.</p>
<p>I’ve tried this, but it doesn’t seem to work:</p>
<pre><code class="lang-auto">on:
  create:
    tags-ignore: ['**']
</code></pre>

**Comment by SoftCreatR** [comment]
<p>For me, this works:</p>
<pre><code class="lang-auto">on:
  push:
    tags-ignore:
      - '**'
</code></pre>

**Comment by starikcetin** [comment]
This works as expected:
```yml
on:
  push:
    branches:
      - "**"
    tags:
      - "!**"
```
